theano/gof/opt.py

```diff
@@ -333,10 +333,6 @@ def print_profile(stream, prof, level=0):
                 print(blanc, '  %.6fs - %s' % (t, opt), file=stream)
 
             if sub_profs[i]:
-                if isinstance(opts[i], MergeOptimizer):
-                    if len(sub_profs[i]) == 8:
-                        sub_profs[i] = sub_profs[i][:-1]
-                    assert len(sub_profs[i]) == 7
```

Comment:
I could have used the fix which Fred used in the other PR, but that didn't strike me